dev/mima

```diff
@@ -42,7 +42,7 @@ $JAVA_CMD \
   -cp "$TOOLS_CLASSPATH:$OLD_DEPS_CLASSPATH" \
   org.apache.spark.tools.GenerateMIMAIgnore
 
-echo -e "q\n" | build/sbt -mem 4096 -DcopyDependencies=false "$@" mimaReportBinaryIssues | grep -v -e "info.*Resolving"
+echo -e "q\n" | build/sbt -mem 4196 -DcopyDependencies=false "$@" mimaReportBinaryIssues | grep -v -e "info.*Resolving"
```

Comment:
Add more 100m mem to avoid:

```
java.lang.OutOfMemoryError: GC overhead limit exceeded
	at java.lang.StringBuilder.toString(StringBuilder.java:412)
	at com.typesafe.tools.mima.core.MemberInfo.fullName(MemberInfo.scala:16)
	at com.typesafe.tools.mima.core.Problem.matchName(Problems.scala:19)
	at com.typesafe.tools.mima.core.ProblemFilters$ExcludeByName.apply(ProblemFilters.scala:12)
	at com.typesafe.tools.mima.core.ProblemFilters$ExcludeByName.apply(ProblemFilters.scala:8)
	at com.typesafe.tools.mima.core.ProblemReporting$.$anonfun$isReported$2(ProblemReporting.scala:27)
	at com.typesafe.tools.mima.core.ProblemReporting$.$anonfun$isReported$2$adapted(ProblemReporting.scala:27)
	at com.typesafe.tools.mima.core.ProblemReporting$$$Lambda$5945/1166075121.apply(Unknown Source)
	at scala.collection.Iterator.forall(Iterator.scala:955)
	at scala.collection.Iterator.forall$(Iterator.scala:953)
	at scala.collection.Iterator$ConcatIterator.forall(Iterator.scala:179)
	at com.typesafe.tools.mima.core.ProblemReporting$.isReported(ProblemReporting.scala:27)
	at com.typesafe.tools.mima.plugin.SbtMima$.isReported$1(SbtMima.scala:61)
	at com.typesafe.tools.mima.plugin.SbtMima$.$anonfun$reportModuleErrors$3(SbtMima.scala:70)
	at com.typesafe.tools.mima.plugin.SbtMima$.$anonfun$reportModuleErrors$3$adapted(SbtMima.scala:70)
	at com.typesafe.tools.mima.plugin.SbtMima$$$Lambda$5939/1088203167.apply(Unknown Source)
	at scala.collection.TraversableLike.noneIn$1(TraversableLike.scala:319)
	at scala.collection.TraversableLike.filterImpl(TraversableLike.scala:385)
	at scala.collection.TraversableLike.filterImpl$(TraversableLike.scala:297)
	at scala.collection.AbstractTraversable.filterImpl(Traversable.scala:108)
	at scala.collection.TraversableLike.filter(TraversableLike.scala:395)
	at scala.collection.TraversableLike.filter$(TraversableLike.scala:395)
	at scala.collection.AbstractTraversable.filter(Traversable.scala:108)
	at com.typesafe.tools.mima.plugin.SbtMima$.reportModuleErrors(SbtMima.scala:70)
	at com.typesafe.tools.mima.plugin.MimaPlugin$.$anonfun$projectSettings$2(MimaPlugin.scala:36)
	at com.typesafe.tools.mima.plugin.MimaPlugin$.$anonfun$projectSettings$2$adapted(MimaPlugin.scala:26)
	at com.typesafe.tools.mima.plugin.MimaPlugin$$$Lambda$5178/1097580344.apply(Unknown Source)
	at scala.collection.Iterator.foreach(Iterator.scala:943)
	at scala.collection.Iterator.foreach$(Iterator.scala:943)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1431)
	at com.typesafe.tools.mima.plugin.MimaPlugin$.$anonfun$projectSettings$1(MimaPlugin.scala:26)
	at com.typesafe.tools.mima.plugin.MimaPlugin$.$anonfun$projectSettings$1$adapted(MimaPlugin.scala:25)
[error] [launcher] error during sbt launcher: java.lang.OutOfMemoryError: GC overhead limit exceeded
```

Reply:
@LuciferYang just to double check, did this happen very frequently? I wonder this memory setting makes our CI flaky

Reply:
Does the GA instance still allow us to add more memory? If possible, we should continue to increase.

I give https://github.com/apache/spark/pull/40862 to test on GA

If we cannot add more memory, we need to roll back this patch and refactor the mima check mechanism

Reply:
In the previous local testing, `GC overhead limit exceeded` not occur again for mem 4196, but I see that GA still has this issue.